dashboard/app/controllers/test_controller.rb

```diff
@@ -175,4 +190,127 @@ def destroy_level
+      'csp_which_grades': ['11', '12'],
+      'csp_how_offer': 'As an AP course',
+      'csd_which_grades': ['6', '7'],
+      'csa_which_grades': ['11', '12'],
+      'csa_how_offer': 'As an AP course',
+      'csa_phone_screen': 'Yes',
+      'csa_already_know': 'Yes',
+      'replace_existing': 'No, this course will be added to the schedule in addition to an existing computer science course'
+    }
+  end
+
+  def create_teacher_application
+    return unless (user = current_user)
+    regional_partner = RegionalPartner.create!(name: "regional-partner#{Time.now.to_i}-#{rand(1_000_000)}")
+
+    RegionalPartnerProgramManager.create!(program_manager_id: user.id, regional_partner_id: regional_partner.id)
+
+    teacher_name = "teacher#{Time.now.to_i}-#{rand(1_000_000)}"
+    teacher_email = "teacher#{Time.now.to_i}-#{rand(1_000_000)}@test.xx"
+    password = teacher_name + "password"
+    attributes = {
+      name: teacher_name,
+      email: teacher_email,
+      password: password,
+      user_type: "teacher",
+      age: "21+"
+    }
+    teacher = User.create!(attributes)
+
+    form_data = teacher_form_data.merge(
+      first_name: teacher_name,
+      last_name: 'Test',
+      program: Pd::Application::TeacherApplication::PROGRAMS[:csp]
+    ).to_json
+    application = Pd::Application::TeacherApplication.create!(
+      user: teacher,
+      form_data: form_data,
+      regional_partner_id: regional_partner.id,
+      course: 'csp',
+      status: 'unreviewed'
+    )
+
+    render json: {rp_id: regional_partner.id, teacher_id: teacher.id, application_id: application.id}
+  end
+
+  def create_applications
+    %w(csd csp csa).each do |course|
+      (Pd::Application::TeacherApplication.statuses).each do |status|
+        teacher_email = "#{course}_#{status}@code.org"
+        teacher = User.find_or_create_teacher(
+          {name: "#{course} #{status}", email: teacher_email}, nil, nil
+        )
+        next if Pd::Application::TeacherApplication.find_by(
+          application_year: Pd::Application::ActiveApplicationModels::APPLICATION_CURRENT_YEAR,
+          user_id: teacher.id
+        )
+
+        form_data = teacher_form_data.merge(
+          first_name: course,
+          last_name: status,
+          program: Pd::Application::TeacherApplication::PROGRAMS[course.to_sym]
+        ).to_json
+
+        if status == 'incomplete'
+          Pd::Application::TeacherApplication.create!(
+            form_data: form_data,
+            user: teacher,
+            course: course,
+            status: 'incomplete'
+          )
+        else
+          application = Pd::Application::TeacherApplication.create!(
+            form_data: form_data,
+            user: teacher,
+            course: course,
+            status: 'unreviewed'
+          )
+          application.update!(status: status)
+        end
+      end
+    end
+    head :ok
+  end
+
+  def delete_rp_pm_teacher_application
+    User.find_by(name: params[:pm_name]).destroy
```

Comment:
if it's too much work, nbd, but is there a way to make this `user_name` instead of `name`? I know you've put a lot of effort into making sure the `name` is unique so this isn't blocking

Reply:
I'm trying to trace how this works, but is `username` better because that's what gets saved? I don't really see where the magic happens in https://github.com/code-dot-org/code-dot-org/blob/staging-next/dashboard/test/ui/features/step_definitions/account_steps.rb#L54

BUT it does look like there's https://github.com/code-dot-org/code-dot-org/blob/staging-next/dashboard/test/ui/features/step_definitions/account_steps.rb#L65 which might be better? I'll try both!

Reply:
`username` is better because it is unique across all rows, unlike `name` (plus it migh. but because you've randomized `name` so much, it's not a big deal

Reply:
Ah, yes, that makes sense and will be better! I can't really tell exactly how `name` is getting used in the helper functions –– I'm not convinced it maps to the `name` field in the DB. (And the method in account_steps requires the rails environment so that's out.)

Reply:
Confirmed that `user_name` didn't find the PM, but `name` does –– sticking with `name` here